### ISSUE TITLE
[xwayland] If you write a capital letter in russian, you get a capital english

### ISSUE THREAD

**Issue by steepboy** [hyprwm/Hyprland#7021]
### Regression?

Yes

### System Info and Version

<details> <summary>System/Version info</summary>

```sh
Hyprland, built from branch main at commit 3c758db95c129ed6ca7ce0c1b5b82ad6e189488d  (renderer/layer-shell: use explicitly set exclusiveEdge (6984)).
Date: Tue Jul 23 00:36:58 2024
Tag: v0.41.2-77-g3c758db9, commits: 4963

flags: (if any)


System Information:
System name: Linux
Node name: yiski-arch
Release: 6.10.0-arch1-2
Version: #1 SMP PREEMPT_DYNAMIC Mon, 22 Jul 2024 17:28:23 +0000


GPU information:
00:02.0 VGA compatible controller [0300]: Intel Corporation CoffeeLake-H GT2 [UHD Graphics 630] [8086:3e9b] (prog-if 00 [VGA controller])
	DeviceName: Second VGA
	DeviceName: VGA


os-release: NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://gitlab.archlinux.org/groups/archlinux/-/issues"
PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
LOGO=archlinux-logo


plugins:
```

</details>

### Description

When writing a capital letter on the russian layout, on all applications running on Xwayland, instead of a capital russian letter, the result is an english one (video attached below)

### How to reproduce

Try to write something on the russian keyboard with shift held down in an application working in Xwayland

### Crash reports, logs, images, videos

https://github.com/user-attachments/assets/b0493156-db47-4f71-9e99-3b9236be24e2

**Comment by sub-kek** [comment]
Caps lock doesn't work either.

**Comment by vaxerski** [comment]
that's already reported

**Comment by sub-kek** [comment]
@steepboy я тебя ~~люблю~~

**Comment by steepboy** [comment]
@sub-kek готовь попку, я захожу)

**Comment by sub-kek** [comment]
> @sub-kek готовь попку, я захожу)

Жду в войсе 💕

**Comment by vaxerski** [comment]
can you two stop spamming please?

**Comment by steepboy** [comment]
fixed on cd942ad12d9d2030963e3dd5fbfe2d181013c750
